src/daft-core/src/record_batch.rs

```diff
@@ -0,0 +1,57 @@
+use std::sync::Arc;
+
+use common_error::{DaftError, DaftResult};
+use daft_schema::schema::{Schema, SchemaRef};
+use serde::{Deserialize, Serialize};
+
+use crate::series::Series;
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+/// A RecordBatch is a collection of columns that have the same length.
+pub struct RecordBatch {
```

Comment:
I think it'd make a lot of sense to instead use a `Table` here, but `daft-table` pulls in a bunch of dependencies we don`t want in core. If we can clean up those dependencies, I think it'd be ideal to use `Table` instead.